specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/scenarios/applicationGateway.yaml

```diff
@@ -0,0 +1,37 @@
+scope: ResourceGroup
+
+variables:
+  userAssignedIdentities:
+    type: string
+
+prepareSteps:
+  - step: Generate_Unique_Name
+    armTemplate: ./peArmTemplate/Generate_Unique_name_template.json
```

Comment:
To generate resource name, please use `variable prefix` instead. Reference: https://github.com/Azure/azure-rest-api-specs-pr/blob/216bf111f8c672fd4e75ccfa02ce2140271d8150/specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/scenarios/vm_quickstart.yaml#L5-L7

Reply:
@leni-msft What does prefix represent for? I could not find the definition from doc.

Reply:
With `prefix`, it means the consumer need generate string value with the prefix. In oav, a string with regex `[0-9a-z]{6}` (all lower case, with length of 6) is generated at runtime.

Reply:
@changlong-liu mentioned this is aligned with current sdk testing framework.